CONTRIBUTING.md

```diff
@@ -0,0 +1,167 @@
+perltidy -l=99 -conv -ci=4 -ola -ce -nbbc -kis -pt=2 -b ddclient
+```
+
+## Git Hygiene
+
+  * Please keep your pull request commits rebased on top of master.
+  * Please use `git rebase -i` to make your commits easy to review:
+    - Put unrelated changes in separate commits
+    - Squash you fixup commits
+  * Write your commit message in imperative mood, and explain *why*
+    the change is made (unless obvious) in addition to *what* is
+    changed.
+
+If you are not very comfortable with Git, we encourage you to read
+[Pro Git](https://git-scm.com/book) by Scott Chacon and Ben Straub
+(freely available online).
+
+## Compatibility
+
+We strive to make ddclient usable on a wide variety of platforms.
+Please limit yourself to Perl language features and modules available
+on the following platforms:
+
+  * Debian oldstable and newer
+  * Ubuntu, [all maintained
+    releases](https://ubuntu.com/about/release-cycle)
+  * Fedora, [all maintained
+    releases](https://fedoraproject.org/wiki/Fedora_Release_Life_Cycle)
+  * CentOS, [all maintained
+    releases](https://wiki.centos.org/About/Product)
+  * Red Hat Enterprise Linux, [all maintained
+    releases](https://access.redhat.com/support/policy/updates/errata/)
+
+See https://pkgs.org for available modules and versions.
+
+Exception: You may depend on modern language features or modules for
+new functionality when no feasible alternative exists, as long as the
+new dependency does not break existing functionality on old plaforms.
+
+All shell scripts should conform with [POSIX Issue 7 (2018
+edition)](https://pubs.opengroup.org/onlinepubs/9699919799/) or later.
+
+## For ddclient Project Maintainers
+
+### Merging Pull Requests
+
+To facilitate reviews and code archaeology, `master` should have a
+semi-linear commit history like this:
+
+```
+*   f4e6e90 sandro.jaeckel@gmail.com 2020-05-31 07:29:51 +0200 (master)
+|\          Merge pull request #142 from rhansen/config-line-format
+| * 30180ed rhansen@rhansen.org 2020-05-30 13:09:38 -0400
+|/          Expand comment documenting config line format
+*   01a746c rhansen@rhansen.org 2020-05-30 23:47:54 -0400
+|\          Merge pull request #138 from rhansen/dyndns-za-net
+| * 08c2b6c rhansen@rhansen.org 2020-05-29 14:44:57 -0400
+|/          Replace dydns.za.net with dyndns.za.net
+*   d65805b rhansen@rhansen.org 2020-05-30 22:30:04 -0400
+|\          Merge pull request #140 from ddclient/fix-interpolation
+| * babbef1 sandro.jaeckel@gmail.com 2020-05-30 04:03:44 +0200
+|/          Fix here doc interpolation
+*   6ae69a1 rhansen@rhansen.org 2020-05-30 22:23:57 -0400
+|\          Merge pull request #141 from ddclient/show-debug-ssl
+| * 096288e sandro.jaeckel@gmail.com 2020-05-30 04:42:27 +0200
+| |         Expand tabs to spaces in vim
+| * 0206262 sandro.jaeckel@gmail.com 2020-05-30 04:40:58 +0200
+|/          Show debug connection settings after evaluating use-ssl
+...
+```
+
+See https://stackoverflow.com/a/15721436 for an explanation of the
```

Comment:
I don't believe this is an useful approch if there's only one commit.  Correct me if I'm wrong.

Reply:
It's less useful for sure, but still has value. In particular, the merge commit gets a reference to the pull request. (I think that the "squash and merge" option modifies the commit description to add a reference to the pull request, which would make it even less useful.)

Reply:
Done.